### PR TITLE
add log2Int function to prelude

### PR DESCRIPTION
**Pull request by dwightguth**
This is much faster than bitshifting manually, which is why I added it.

### CODE REVIEW

**Comment by ehildenb** [comment]
@dwightguth can you add an implementation for the Java backend as well?

**Comment by ehildenb** [comment]
I'm testing with KEVM now.

**Comment by ehildenb** [comment]
Cannot approve until we pass tests on https://github.com/kframework/evm-semantics/pull/222.

File: java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIntOperations.java [hunk 1]
```diff
@@ -99,6 +99,18 @@ public static IntToken abs(IntToken term, TermContext context) {
         return IntToken.of(term.bigIntegerValue().abs());
     }
 
+    public static IntToken log2(IntToken term, TermContext context) {
```
Comment (traiansf) [inline java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIntOperations.java:102]: Code seems legitimate to me.  However, why wouldn't `bitLength()` work here?

File: k-distribution/include/ocaml/prelude.ml [hunk 2]
```diff
@@ -835,6 +835,9 @@ struct
   let hook_min c _ _ _ _ = match c with
       [Int a], [Int b] -> [Int (Z.min a b)]
     | _ -> raise Not_implemented
+  let hook_log2 c _ _ _ _ = match c with
```
Comment (traiansf) [inline k-distribution/include/ocaml/prelude.ml:838]: similarly here, it seems you could be faster by using number of limbs, limb size, and computing the bit length of the most significant limb.

File: java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIntOperations.java (hunk 1 above)
Comment (dwightguth) [reply java-backend/src/main/java/org/kframework/backend/java/builtins/BuiltinIntOperations.java:102]: bitLength ceilings the log2, which is not the behavior we want, we want it floored.

File: k-distribution/include/ocaml/prelude.ml (hunk 2 above)
Comment (dwightguth) [reply k-distribution/include/ocaml/prelude.ml:838]: The performance is more than good enough with this as is.

Comment (dwightguth) [reply k-distribution/include/ocaml/prelude.ml:838]: I could maybe do what you suggest but it would be more fragile, so I'm inclined to leave it.

**Review by ehildenb — APPROVED** [review]
KEVM tests pass, which exercises this code path. Still want to hear from @traiansf on the issues above.

**Review by traiansf — APPROVED** [review]
If there is no issue of performance, I have no further comments.

**Pull request merged**